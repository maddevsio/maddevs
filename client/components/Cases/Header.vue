<template>
  <section class="case_header">
    <video class="case_main-video" loop="true" muted="true" autoplay="true">
      <source src="../../assets/img/Cases/nambafood/video/main.mp4" type="video/mp4">
      Your browser does not support the video tag.
    </video>
    <div class="case_logotype nambafood"></div>
    <h1 class="case_title">Namba Food <br> Top Delivery Service <br class="case_text-break"> in <br class="case_text-break-mobile"> Central Asia</h1>
    <button type="button" class="case_more__button">Read our case studies â†“</button>
  </section>
</template>

<script>
export default {
  name: 'Header'
};
</script>

<style scoped lang="scss">
@import "../../assets/styles/cases/header";
@import "../../assets/styles/cases/mixins";

.case {
  &_main-video {
    position: absolute;
    top: 0;
    left: 0;
    z-index: -1;
    height: 100%;
    width: 100%;
    object-fit: cover;
  }

  &_header {
    height: 100vh;
    min-height: 568px;
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
    position: relative;

    &:before {
      content: '';
      width: 100%;
      height: 100%;
      position: absolute;
      top: 0;
      left: 0;
      background-image: url("../../assets/img/Cases/nambafood/jpg/blank.jpg");
      background-repeat: no-repeat;
      background-size: cover;
      z-index: -1;
    }

    &:after {
      content: '';
      display: block;
      width: 100%;
      height: 100%;
      position: absolute;
      top: 0;
      left: 0;
      background: rgba(0, 0, 0, .2);
    }
  }

  &_logotype {
    @include nambafood-header;
    width: 127px;
    height: 114px;
    margin-bottom: 54px;
    z-index: 1;
  }

  &_title {
    @include title($text-color--white, 51.88px, -2px);
    text-align: center;
    z-index: 1;
  }

  &_more__button {
    position: absolute;
    bottom: 0;
    max-width: 1104px;
    height: 55px;
    width: 100%;
    background-color: $bgcolor--pink;
    border: 0;
    z-index: 1;
    cursor: pointer;

    @include default_text($text-color--white, 18px, 30px, -0.035em, normal);
  }

  &_text-break-mobile {
    display: none;
  }

  @media screen and (max-width: 768px) {
    &_title {
      @include title($text-color--white, 35px, -1px);
    }

    &_more__button {
      @include default_text($text-color--white, 16px, 150%, -0.02em, normal);
    }

    &_logotype {
      width: 103.51px;
      height: 92.75px;
      margin-bottom: 46px;
    }
  }

  @media screen and (max-width: 440px) {
    &_title {
      max-width: 290px;
    }

    &_text-break {
      display: none;
    }

    &_text-break-mobile {
      display: block;
    }
  }
}
</style>
