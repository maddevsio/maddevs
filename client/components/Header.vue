<template>
  <div class="header-wrapper">
    <div class="overlay" v-if="headerTransparent" ref="overlay"></div>
    <header ref="header" class="header" :class="{'header-transparent': headerTransparent}">
      <div class="container" ref="headerContainer">
        <div class="row">
          <div class="header__left-nav_bar col-xl-6 col-lg-7">
            <router-link :to="`/`" class="header__logo-icon">
              <headerLogo class="header__header-logo" :isCasePage="isCasePage"/>
            </router-link>
            <nav class="header__header-routes_links">
              <router-link @click.native="goToTopPage" exact to="/" class="header__navigation-link">About</router-link>
              <router-link @click.native="goToTopPage" to="/services" class="header__navigation-link">Services</router-link>
              <router-link @click.native="goToTopPage" to="/projects" class="header__navigation-link">Projects</router-link>
              <router-link @click.native="goToTopPage" to="/careers" class="header__navigation-link">Careers</router-link>
              <a href="https://blog.maddevs.io/" target="_blank" rel="noreferrer" class="header__navigation-link header__navigation-link-blog">Blog</a>
            </nav>
          </div>
          <div class="header__right-content col-xl-6 col-lg-5">
            <div class="header__phone-wrapper">
              <span class="header__phone-flag" />
              <a class="header__header-phone" href="tel:+44 20 3984 8555">+44 20 3984 8555</a>
            </div>
            <UIButtonModalTrigger
              :buttonInnerText="buttonInnerText"
              :modalWindowName="modalWindowName"
              :isRed="true"
            />
          </div>
        </div>
      </div>
    </header>
    <mobileHeader/>
  </div>
</template>

<script>
import UIButtonModalTrigger from '@/components/ui/UIButtonModalTrigger';
import mobileHeader from '@/components/Header/mobile-header';
import headerLogo from '@/components/svg/headerLogo';

export default {
  name: 'main-header',
  components: {
    UIButtonModalTrigger,
    mobileHeader,
    headerLogo
  },
  data() {
    return {
      buttonInnerText: 'Contact me',
      selectedPhone: null,
      modalWindowName: 'contact-me-modal',
      headerTransparent: false,
      scrollTop: null,
      isCasePage: false,
      caseMoreButton: null,
      caseHeader: null,
      logoText: null
    };
  },
  created() {
    if(this.$nuxt.$route.name === 'case-studies-namba-food') {
      this.headerTransparent = true;
      this.isCasePage = true;
    } else {
      this.headerTransparent = false;
      this.isCasePage = false;
    }
  },
  mounted() {
    if(this.isCasePage) {
      this.caseMoreButton = document.getElementsByClassName('case_more__button')[0];
      this.caseHeader = document.getElementsByClassName('case_header')[0];
      this.logoText = document.getElementsByClassName('header-logo-text')[0];

      this.getScrollTop();
      window.addEventListener('scroll', () => {
        this.scrollHandler();
      });
    }
  },
  watch: {
    '$route'() {
      this.headerTransparent = false;
      this.isCasePage = false;
    }
  },
  methods: {
    goToTopPage() {
      window.scrollTo(0, 0);
    },
    getScrollTop() {
      this.scrollTop = this.caseMoreButton.getBoundingClientRect().top - this.$refs.headerContainer.offsetHeight;
    },
    scrollHandler() {
      if(this.isCasePage && window.innerWidth > 991) {
        const opacity = 1.6 - (this.$refs.overlay.offsetHeight - (window.scrollY - this.caseHeader.getBoundingClientRect().height + this.caseMoreButton.getBoundingClientRect().height) - this.$refs.headerContainer.offsetHeight) / this.$refs.overlay.offsetHeight;
        const opacityTextLogo = 0.9 - (this.$refs.overlay.offsetHeight - this.caseMoreButton.getBoundingClientRect().top + this.caseMoreButton.getBoundingClientRect().height) / this.$refs.overlay.offsetHeight;
        this.$refs.overlay.style.opacity = opacity;
        this.logoText.style.opacity = opacityTextLogo;
      } else if(this.isCasePage && window.innerWidth < 991) {
        const opacity = 1.7 - (this.$refs.overlay.offsetHeight - (window.scrollY - this.caseHeader.getBoundingClientRect().height + this.caseMoreButton.getBoundingClientRect().height) - 60) / this.$refs.overlay.offsetHeight;
        this.$refs.overlay.style.opacity = opacity;
      }
    }
  }
};
</script>

<style lang="scss" scoped>
  @import '../assets/styles/vars';
  @import '../assets/styles/_flagsIcons';

  .header {
    width: 100%;
    height: 40px;
    padding: 11px 0;
    position: fixed;
    z-index: 2;
    background-color: $bgcolor--black;

    &__header-logo {
      width: 34px;
      height: 58px;
      margin: 5px -33px 0 -65px;
    }

    &__left-nav_bar,
    &__right-content {
      display: flex;
      align-items: center;
    }

    &__right-content {
      justify-content: flex-end;
    }

    &__phone-wrapper,
    &__header-routes_links,
    button {
      margin-bottom: 29px;
    }

    button {
      width: 132px;
      height: 39px;
      border-radius: 4px;
      border-color: $text-color--red-opacity;
    }

    &__navigation-link,
    &__header-phone {
      color: $text-color--white;
    }

    &__navigation-link,
    &__header-phone,
    button {
      font-size: 16px;
      font-family: 'Poppins-Regular', sans-serif;
      text-decoration: none;
      letter-spacing: -0.02em;
    }

    &__navigation-link {
      margin-right: 15px;

      &::after {
        content: 'â†“';
        font-size: 17px;
        font-family: 'Poppins-Regular', sans-serif;;
        color: transparent;
      }
    }

    &__navigation-link-blog {
      &::after {
        content: '';
      }
    }

    .nuxt-link-active {
      color: $text-color--red;

      &::after {
        color: $text-color--red;
      }
    }

    &__phone-wrapper {
      display: flex;
      align-items: center;
      margin-right: 40px;

      @media screen and (max-width: 1366px) {
        margin-right: 49px;
      }

      @media screen and (max-width: 1280px) {
        margin-right: 33px;
      }

      @media screen and (max-width: 1024px) {
        margin-right: 20px;
      }
    }

    &__header-phone {
      display: flex;
      align-items: center;
      position: relative;
      margin-left: 7px;
      cursor: pointer;

      img {
        position: absolute;
        top: 4px;
        left: 0;
      }
    }

    &__phone-flag {
      display: block;
      width: 18px;
      height: 14px;
      @include flag-uk;
    }

    @media screen and (max-width: 991px) {
      display: none;
    }
  }

  .header-transparent {
    background-color: transparent !important;
  }

  .overlay {
    width: 100%;
    height: 40px;
    padding: 11px 0;
    position: fixed;
    z-index: 2;
    background-color: $bgcolor--black;
    opacity: 0;
    transition: opacity 0.6s;

    @media screen and (max-width: 991px) {
      height: 48px;
      padding: 0;
    }
  }

  .header-default {
    background-color: $bgcolor--black !important;

    &::before {
      opacity: 0;
    }
  }

  .mobile-menu_is-open {
    width: 100%;
    height: 100%;
    position: fixed;
    padding: initial;
    z-index: 999;
  }

</style>
