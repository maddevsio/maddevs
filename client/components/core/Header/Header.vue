<template>
  <div
    id="header"
    class="header-wrapper"
    data-testid="test-header-wrapper"
    :class="{ 'is-transparent-bg': isTransparentBG }"
  >
    <div
      id="overlay"
      ref="overlay"
    />
    <header
      ref="header"
      data-testid="test-header"
      :class="{ transparent: isCasePage && !isActiveMobileMenu }"
      class="header"
    >
      <div
        ref="headerContainer"
        class="container"
      >
        <div class="row">
          <div class="header__left-nav_bar col-xl-6 col-lg-7">
            <NuxtLink
              :to="`/`"
              class="header__logo-icon"
            >
              <HeaderLogo
                :is-show-text="showLogoText"
                :is-case-page="isCasePage"
                :is-active-mobile-menu="isActiveMobileMenu"
                class="header__header-logo"
              />
            </NuxtLink>
            <nav class="header__header-routes_links">
              <NuxtLink
                v-for="{ title, link, exact } in navigation"
                :key="link"
                :exact="exact"
                class="header__navigation-link"
                data-testid="test-navigation-link"
                :to="link"
                @click.native="goToTopPage"
              >
                {{ title }}
              </NuxtLink>
            </nav>
            <!-- Burget btn -->
            <div
              class="header__burger"
              data-testid="test-burger"
              @click="toggleMobileMenu"
            >
              <svg
                v-if="isActiveMobileMenu"
                class="header__burger--close"
                data-testid="test-burger--close"
                width="20"
                height="20"
                viewBox="0 0 20 20"
                fill="none"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0.807613 19.1924L19.1924 0.807623M19.1914 19.1924L0.806641 0.807617"
                  stroke="#fff"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
              <svg
                v-else
                class="header__burger--open"
                width="30"
                height="10"
                viewBox="0 0 30 10"
                fill="none"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0H30V1H0V0Z"
                  fill="#F5F7F9"
                />
                <path
                  d="M0 9H30V10H0V9Z"
                  fill="#F5F7F9"
                />
              </svg>
            </div>
            <!-- END Burget btn -->
          </div>
          <div class="header__right-content col-xl-6 col-lg-5">
            <div class="header__phone-wrapper">
              <!-- Flag uk -->
              <img
                width="19"
                height="14"
                src="@/assets/img/Home/flags/uk.svg"
                class="header__phone-flag"
                alt="United Kingdom"
              >
              <a
                class="header__header-phone"
                href="tel:+442039848555"
              >+44 20 3984 8555</a>
            </div>
            <UIModalTriggerButton
              label="Contact me"
              color="red"
              @click="$refs.modalContactMe.show()"
            />
          </div>
        </div>
      </div>
    </header>

    <!-- Mobile header -->
    <HeaderMobile
      v-if="isActiveMobileMenu"
      :enable-page-scroll="enablePageScroll"
      @changed-page="isActiveMobileMenu = false"
      @open-modal="$refs.modalContactMe.show()"
    />
    <!-- END Mobile header -->

    <Modal ref="modalContactMe">
      <ModalContactMe @success="$refs.modalContactMe.close()" />
    </Modal>
  </div>
</template>

<script>
import UIModalTriggerButton from '@/components/shared/UIModalTriggerButton'
import HeaderMobile from '@/components/core/Header/HeaderMobile'
import HeaderLogo from '@/components/core/Header/HeaderLogo'
import Modal from '@/components/core/Modal'
import { headerNavigation as navigation } from '@/data/navigation'

export default {
  name: 'MainHeader',
  components: {
    UIModalTriggerButton,
    ModalContactMe: () => import('@/components/core/modals/ModalContactMe'),
    HeaderMobile,
    HeaderLogo,
    Modal,
  },

  data() {
    return {
      navigation,
      showLogoText: true,
      isActiveMobileMenu: false,
      isCasePage: false,
      isTransparentBG: true,
      caseGoDeeScrollContainer: null,
    }
  },

  computed: {
    isGodeePage() {
      return this.$nuxt.$route.path.includes('/godee')
    },
  },

  watch: {
    $route() {
      this.setDefaultStateForHeader()
      this.removeEventListeners()
    },
  },

  created() {
    this.setDefaultStateForHeader()
  },

  mounted() {
    this.caseGoDeeScrollContainer = document.getElementById('case-scroll-container')
    this.addEventListeners()
    this.setStylesForHeader()
    if (this.isCasePage && this.isGodeePage) this.setWidthForHeader()
  },

  methods: {
    // Base methods
    goToTopPage() {
      window.scrollTo(0, 0)
    },

    preventDefault(e) {
      e.preventDefault()
    },

    enablePageScroll() {
      const scrollY = document.body.style.top || '0'
      document.body.removeEventListener('touchmove', this.preventDefault)
      document.body.classList.remove('scrollDisabled')
      document.documentElement.classList.remove('scrollDisabled')
      window.scrollTo(0, parseInt(scrollY, 10) * -1)
    },

    disablePageScroll() {
      const { scrollY } = window
      document.body.addEventListener('touchmove', this.preventDefault, { passive: false })
      document.body.classList.add('scrollDisabled')
      document.documentElement.classList.add('scrollDisabled')
      document.body.style.top = `-${scrollY}px`
    },

    setDefaultStateForHeader() {
      this.isCasePage = this.$nuxt.$route.path.includes('/case-studies/')
    },

    changeLogoState(scrollTop) {
      this.showLogoText = Boolean(scrollTop < 10)
    },

    handleMobileMenuScroll(e) {
      e.stopImmediatePropagation()
      this.changeLogoState(this.mobileHeaderScrollbar.scrollTop)
    },

    toggleMobileMenu() {
      this.isActiveMobileMenu = !this.isActiveMobileMenu
      if (this.isActiveMobileMenu) {
        this.disablePageScroll()
        this.$nextTick(() => {
          this.mobileHeaderScrollbar = document.getElementById('mobile-header-scrollbar')
          this.mobileHeaderScrollbar.addEventListener('scroll', this.handleMobileMenuScroll)
          this.mobileHeaderScrollbar.addEventListener('touchmove', this.handleMobileMenuScroll)
        })
      } else {
        this.enablePageScroll()
        this.mobileHeaderScrollbar.removeEventListener('scroll', this.handleMobileMenuScroll)
        this.mobileHeaderScrollbar.removeEventListener('touchmove', this.handleMobileMenuScroll)
      }
    },

    setWidthForHeader() {
      if (!this.caseGoDeeScrollContainer) return
      const scrollBarWidth = this.caseGoDeeScrollContainer.offsetWidth - this.caseGoDeeScrollContainer.clientWidth
      if (this.$refs.header && this.$refs.overlay) {
        if (window.innerWidth >= 991) {
          this.$refs.header.style.width = `calc(100% - ${scrollBarWidth}px)`
          this.$refs.overlay.style.width = `calc(100% - ${scrollBarWidth}px)`
        } else {
          this.$refs.header.style.width = '100%'
          this.$refs.overlay.style.width = '100%'
        }
      }
    },

    setStylesForHeader() {
      const scrollTop = this.isGodeePage && this.caseGoDeeScrollContainer
        ? this.caseGoDeeScrollContainer.scrollTop
        : window.scrollY
      const area = document.querySelector('#case-header')
      if (!area) return

      const areaHeight = (area.offsetTop + area.offsetHeight) - (this.$refs.overlay && this.$refs.overlay.offsetHeight)
      // const isAfterTopPointSection = scrollTop >= area.offsetTop // After Top point of the section
      const isBeforeBottomPointSection = scrollTop <= areaHeight // Before Bottom point of the section

      this.isTransparentBG = isBeforeBottomPointSection
    },

    scrollHandler() {
      if (this.$nuxt.$route.path.includes('/case-studies/')) this.setStylesForHeader()
      this.changeLogoState(window.pageYOffset)
    },

    addEventListeners() {
      window.addEventListener('scroll', this.scrollHandler)
      if (!this.isGodeePage) return

      window.addEventListener('resize', this.setWidthForHeader)
      if (this.caseGoDeeScrollContainer) this.caseGoDeeScrollContainer.addEventListener('scroll', this.scrollHandler)
    },

    removeEventListeners() {
      if (!this.isGodeePage) return

      window.removeEventListener('resize', this.setWidthForHeader)
      if (this.caseGoDeeScrollContainer) this.caseGoDeeScrollContainer.removeEventListener('scroll', this.scrollHandler)
    },
  },
}
</script>

<style lang="scss" scoped>
@import '../../../assets/styles/vars';

.header {
  width: 100%;
  height: 40px;
  padding: 11px 0;
  position: fixed;
  top: 0;
  left: 0;
  z-index: 3;
  background-color: $bgcolor--black;

  &__burger {
    display: none;
    position: fixed;
    top: 0;
    right: 25px;
    padding: 11px 14px;

    &--close {
      width: 22px;
      height: 22px;
      margin-top: 3px;
      margin-right: 3px;
    }
  }

  &__header-logo {
    width: 34px;
    height: 58px;
    margin: -3px -33px 0 -65px;
  }

  &__left-nav_bar,
  &__right-content {
    display: flex;
    align-items: center;
  }

  &__right-content {
    justify-content: flex-end;
  }

  &__phone-wrapper,
  &__header-routes_links,
  button {
    margin-bottom: 29px;
  }

  button {
    width: 132px;
    height: 39px;
    border-radius: 4px;
    border-color: $text-color--red-opacity;
  }

  &__navigation-link,
  &__header-phone {
    color: $text-color--white;
  }

  &__navigation-link,
  &__header-phone,
  button {
    font-size: 16px;
    font-family: 'Poppins-Regular', sans-serif;
    text-decoration: none;
    letter-spacing: -0.02em;
  }

  &__navigation-link {
    margin-right: 15px;

    &::after {
      content: '↓';
      font-size: 17px;
      font-family: 'Poppins-Regular', sans-serif;
      color: transparent;
    }
  }

  .nuxt-link-active {
    color: $text-color--red;

    &::after {
      color: $text-color--red;
    }
  }

  &__phone-wrapper {
    display: flex;
    align-items: center;
    margin-right: 40px;

    @media screen and (max-width: 1366px) {
      margin-right: 49px;
    }

    @media screen and (max-width: 1280px) {
      margin-right: 33px;
    }

    @media screen and (max-width: 1024px) {
      margin-right: 20px;
    }
  }

  &__header-phone {
    display: flex;
    align-items: center;
    position: relative;
    margin-left: 7px;
    cursor: pointer;

    img {
      position: absolute;
      top: 4px;
      left: 0;
    }
  }

  &__phone-flag {
    display: block;
    width: 18px;
    height: 14px;
  }

  &.transparent {
    background: transparent;
  }
}

// ------------ Overlay styles ------------- //
#overlay {
  width: 100%;
  height: 40px;
  padding: 11px 0;
  position: fixed;
  z-index: 2;

  &::before {
    content: '';
    width: 100%;
    height: 100%;
    position: absolute;
    left: 0;
    top: 0;
    background-color: $bgcolor--black;
    opacity: 1;
  }

  @media screen and (max-width: 991px) {
    height: 48px;
    padding: 0;
  }
}

/deep/ #header-logo-text {
  opacity: 0;
  transform: translateY(-100px) translateX(5%) scale(0.9);
  transition: all 0.1s ease;
}

.is-transparent-bg {
  #overlay::before {
    height: 0;
    opacity: 0;
  }

  /deep/ #header-logo-text {
    opacity: 1;
    transform: none;
  }
}

// ------------ END Overlay styles ------------- //

.mobile-menu_is-open {
  width: 100%;
  height: 100%;
  position: fixed;
  padding: initial;
  z-index: 999;
}

@media screen and (max-width: 991px) {
  .header {
    max-height: 26px;

    &__burger {
      display: block;
    }

    &__left-nav_bar {
      margin-top: 0;
    }

    &__header-routes_links {
      display: none;
    }

    &__right-content {
      display: none;
    }

    &__header-logo {
      width: 28px;
      height: 49px;
      margin-top: 20px;
      margin-left: -55px;
    }
  }
}

@media screen and (max-width: 767px) {
  .header {
    &__burger {
      right: 10px;
    }

    &__header-logo {
      margin-left: -40px;
    }
  }
}

@media screen and (max-width: 576px) {
  .header {
    &__header-logo {
      margin-left: -25px;
    }
  }
}

@media screen and (max-width: 440px) {
  .header {
    &__header-logo {
      margin-left: 0;
    }
  }
}
</style>
