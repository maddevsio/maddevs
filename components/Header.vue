<template>
  <header class="header" :class="{ 'header--dark': headerDark }">
    <div class="container">
      <div class="header-wrap">
        <router-link :to="`/${this.lang}`">
          <SvgLogo class="header-logo"  :isDark="headerDark"/>
        </router-link>
        <div class="header-wrap_right-block">
          <nav class="header_links">
            <a :href="`https://blog.maddevs.io/${this.lang === 'ru' ? 'ru' : ''}`" target="_blank" rel="noreferrer">{{ $t('header-link_blog') }}</a>
            <router-link :to="`/${this.lang}/jobs`">{{ $t('header-link_careers') }}</router-link>
          </nav>
          <div class="header_soc-icons">
            <a href="https://ru.linkedin.com/company/mad-devs" target="_blank" rel="noreferrer">
              <linkedin :isDark="headerDark" />
            </a>
            <a href="https://www.facebook.com/maddevsio" target="_blank" rel="noreferrer">
              <facebook :isDark="headerDark" />
            </a>
          </div>
          <button v-if="lang == 'en'" class="switch-lang" v-on:click.prevent="switchLanguage('ru')">Русский</button>
          <button v-else class="switch-lang" v-on:click.prevent="switchLanguage('en')">English</button>
        </div>
        <button @click="toggleMobileMenu()" class="header-mobile-menu_open">
          <Hamburger :isDark="headerDark" />
        </button>
      </div>
    </div>
    <MobMenu v-if="mobileMenuActive" @CloseMobileMenu="toggleMobileMenu"/>
  </header>
</template>

<script>
import MobMenu from '@/components/ui/mobile-menu';
import linkedin from '@/components/svg/linkedin-icon';
import facebook from '@/components/svg/facebook-icon';
import Hamburger from '@/components/svg/hamburger';
import SvgLogo from '@/components/svg/SvgLogo';

export default {
  name: 'main-header',
  components: {
    MobMenu,
    linkedin,
    facebook,
    Hamburger,
    SvgLogo
  },
  data() {
    return {
      lang: 'en',
      mobileMenuActive: false,
      headerDark: false,
      pagesWithDarkHeader: ['/management', '/vacancies']
    };
  },
  created() {
    this.lang = this.$nuxt.$router.history.current.params.lang === undefined ? 'en' : this.$store.state.locale;
    this.$store.commit('SET_LANG', this.lang);
    if (process.browser) {
      window.addEventListener('resize', this.toggleScrollOnBody);
    }
    this.changeColorHeader();
  },
  watch: {
    '$route' (to, from) {
      this.changeColorHeader();
    }
  },
  beforeDestroy() {
    if (process.browser) {
      window.removeEventListener('resize', this.toggleScrollOnBody);
    }
  },
  methods: {
    changeColorHeader() {
      if (this.pagesWithDarkHeader.includes(this.$nuxt.$router.history.current.path)) {
        this.headerDark = true;
      } else {
        this.headerDark = false;
      }
    },
    switchLanguage(locale) {
      const beforePath = this.$nuxt.$router.history.current.path;
      this.$store.commit('SET_LANG', locale);
      this.lang = this.$store.state.locale;
      let result = '';
      result = beforePath.replace('/en', '');
      result = result.replace('/ru', '');
      if ( locale == 'ru' || locale == 'en' ) {
        this.$nuxt.$router.replace({ path: '/' + locale + result });
      } else {
        if ( result == '/' ) {
          this.$nuxt.$router.replace({ path: '/' + locale });
        } else {
          this.$nuxt.$router.replace({ path: '/' + locale + result });
        }
      }
    },
    toggleMobileMenu(bool) {
      const chat = document.getElementById('tidio-chat');
      if (this.mobileMenuActive === false) {
        this.mobileMenuActive = true;
        this.disableScrollOnBody();
        chat.style.display = 'none';
      } else {
        chat.style.display = 'block';
        this.mobileMenuActive = false;
        this.enableScrollOnBody();
      }
    },
    toggleScrollOnBody() {
      if (window.innerWidth >= 480) {
        this.enableScrollOnBody();
      } else if (window.innerWidth < 480) {
        if (self.mobileMenuActive === true) {
          this.disableScrollOnBody();
        }
      }
    },
    disableScrollOnBody() {
      document.body.classList.add('scrollDisabled');
    },
    enableScrollOnBody() {
      document.body.classList.remove('scrollDisabled');
    }
  }
};
</script>

<style lang="scss" scoped>
  @import '../assets/styles/vars';

  .header {
    width: 100%;
    position: absolute;
    z-index: 2;
    padding-top: 30px;
    &_links {
      position: relative;
      a {
        color: $text-color--white;
        text-decoration: none;
        margin-right: 15px;
        font-size: 16px;
        font-family: 'MADEEvolveSans-regular',
        sans-serif;
      }
      .nuxt-link-active {
        color: $accent-color--red !important;
      }
      .router-link-active {
        color: $accent-color--red !important;
      }
    }
    &-wrap {
      display: flex;
      align-items: flex-start;
      justify-content: space-between;
      &_right-block {
        display: flex;
        align-items: center;
        padding: 24px 0;
        .header_soc-icons {
          height: 22px;
          a {
            display: inline-block;
            margin-right: 10px;

            &:last-child {
              margin-right: 6px;
            }

            img {
              display: block;
            }
          }
        }
      }
    }
    &-logo {
      width: 40px;
      height: 70px;
    }
    .switch-lang {
      background-color: transparent;
      color: $text-color--white;
      font-size: 14px;
      font-family: 'MADEEvolveSans-regular', sans-serif;
      border: none;
      cursor: pointer;
      padding-right: 0;
    }
  }

  .header--dark {
    .container {
      max-width: 100%;

      .header-wrap_right-block {
        .header_links {
          a {
            color: $text-color--black;
          }
        }
      }
    }

    .header-mobile-menu_open {
      svg {
        g {
          fill: $bgcolor--black !important;
        }
      }
    }

    .switch-lang {
      color: $text-color--black !important;
    }
  }

  @media only screen and (min-width: 480px) {
    .header-mobile-menu_open {
      display: none;
    }
  }
  @media only screen and (max-width: 480px) {
    .header_links {
      margin-right: 14px;
      white-space: nowrap;
      a {
        font-size: 14px;
        margin-right: 14px;
      }
    }
    .header-wrap_right-block {
      margin-left: auto;
      padding: 25px;
      .header_links {
        display: none;
      }
      .header_soc-icons {
        display: none;
      }
      .switch-lang {
        font-size: 14px;
      }
    }
    .header-wrap_right-block .header_soc-icons {
      height: 18px;
      white-space: nowrap;
      a {
        margin-right: 12px;
        img {
          width: 18px;
          height: 18px;
        }
      }
    }
    .header-mobile-menu_open {
      display: block;
      background-color: transparent;
      border: none;
      padding: 24px 0;
    }
  }
</style>
