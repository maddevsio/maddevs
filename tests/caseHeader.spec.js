import {
  mount
} from '@vue/test-utils';
import Header from '@/components/Cases/Header';

describe('Header', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(Header);
  });

  // ------ IMPORTANT ----- //
  test('is a Vue instance', () => {
    expect(wrapper.exists()).toBeTruthy();
  });
  
  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
  // --------------------- //
});
