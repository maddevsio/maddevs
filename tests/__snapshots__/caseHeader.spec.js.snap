// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<section
  class="case_header"
>
  <video
    autoplay="autoplay"
    class="case_main-video"
    loop="loop"
    muted="muted"
  >
    <source
      src="../../assets/img/Cases/nambafood/video/main.mp4"
      type="video/mp4"
    />
    
    Your browser does not support the video tag.
  
  </video>
   
  <div
    class="case_logotype nambafood"
  />
   
  <h1
    class="case_title"
  >
    Namba Food 
    <br />
     Top Delivery Service 
    <br
      class="case_text-break"
    />
     in 
    <br
      class="case_text-break-mobile"
    />
     Central Asia
  </h1>
   
  <button
    class="case_more__button"
    type="button"
  >
    Read our case studies â†“
  </button>
</section>
`;
